src/platform/android/DnssdImpl.cpp

```diff
@@ -223,7 +223,7 @@ void HandleResolve(jstring instanceName, jstring serviceType, jstring address, j
         callback(reinterpret_cast<void *>(contextHandle), service, Span<Inet::IPAddress>(), error);
     };
 
-    VerifyOrReturn(address != nullptr && port != 0, dispatch(CHIP_ERROR_UNKNOWN_RESOURCE_ID));
+    VerifyOrReturn(!(address == nullptr || port == 0), dispatch(CHIP_ERROR_UNKNOWN_RESOURCE_ID));
```

Comment:
This change really does not make sense to me.  Could you please try logging the values of the following expressions:
* `address != nullptr`
* `address == nullptr`
* `port != 0`
* `port == 0`
 and reporting what you see?

Reply:
Yes, this time it logged the same, as expected so I reverted this change. It made no sense but it was working with my update and not with the original expression previously. Maybe my other change fixed both or some other change fixed this getting null address and 0 port occasionally...